src/pep257.py

```diff
@@ -679,73 +678,496 @@ def __getattr__(self, item):
+        """Extract the error codes from the selected exclusive option."""
+        codes = set(ErrorRegistry.get_error_codes())
+        checked_codes = None
+
+        if options.ignore is not None:
+            checked_codes = codes - options.ignore
+        elif options.select is not None:
+            checked_codes = options.select
+        elif options.convention is not None:
+            checked_codes = getattr(conventions, options.convention)
+
+        # To not override the conventions nor the options - copy them.
+        return copy.deepcopy(checked_codes)
+
+    @staticmethod
+    def _set_add_options(checked_codes, options):
+        """Set `checked_codes` by the `add_ignore` or `add_select` options."""
+        checked_codes |= options.add_select
+        checked_codes -= options.add_ignore
+
+    @classmethod
+    def _get_checked_errors(cls, options):
+        """Extract the codes needed to be checked from `options`."""
+        checked_codes = cls._get_exclusive_error_codes(options)
+        if checked_codes is None:
+            checked_codes = cls.DEFAULT_CONVENTION
+
+        cls._set_add_options(checked_codes, options)
+
+        return checked_codes - set('')
+
+    @classmethod
+    def _validate_options(cls, options):
+        """Validate the mutually exclusive options.
+
+        Return `True` iff only zero or one of `MUTUALLY_EXCLUSIVE_OPTIONS` was
+        selected.
+
+        """
+        for opt1, opt2 in \
+                itertools.permutations(cls.MUTUALLY_EXCLUSIVE_OPTIONS, 2):
+            if getattr(options, opt1) and getattr(options, opt2):
+                log.error('Cannot pass both {0} and {1}. They are '
+                          'mutually exclusive.'.format(opt1, opt2))
+                return False
+
+        if options.convention and options.convention not in conventions:
+            log.error("Illegal convention '{0}'. Possible conventions: {1}"
+                      .format(options.convention,
+                              ', '.join(conventions.keys())))
+            return False
+        return True
+
+    @classmethod
+    def _has_exclusive_option(cls, options):
+        """Return `True` iff one or more exclusive options were selected."""
+        return any([getattr(options, opt) for opt in
+                    cls.MUTUALLY_EXCLUSIVE_OPTIONS])
+
+    @staticmethod
+    def _fix_set_options(options):
+        """Alter the set options from None/strings to sets in place."""
+        optional_set_options = ('ignore', 'select')
+        mandatory_set_options = ('add_ignore', 'add_select')
+
+        for opt in optional_set_options:
+            value = getattr(options, opt)
+            if value is not None:
+                value = set(value.split(','))
+                setattr(options, opt, value)
+
+        for opt in mandatory_set_options:
+            value = getattr(options, opt)
+            if value is None:
+                value = ''
+
+            if not isinstance(value, set):
```

Comment:
Instead of checking specifically for `set`, check if `isinstance(value, collections.Set)`, so that it checks for the abstract interface.